client/rpcserver/handlers.go

```diff
@@ -385,13 +387,37 @@ func handleCancel(s *RPCServer, params *RawParams) *msgjson.ResponsePayload {
 	if err := s.core.Cancel(form.AppPass, form.OrderID); err != nil {
 		errMsg := fmt.Sprintf("unable to cancel order %q: %v", form.OrderID, err)
 		resErr := msgjson.NewError(msgjson.RPCCancelError, errMsg)
-		return createResponse(tradeRoute, nil, resErr)
+		return createResponse(cancelRoute, nil, resErr)
 	}
 	resp := fmt.Sprintf(canceledOrderStr, form.OrderID)
 
 	return createResponse(cancelRoute, &resp, nil)
 }
 
+// handleWithdraw handles requests for withdraw. *msgjson.ResponsePayload.Error
+// is empty if successful.
+func handleWithdraw(s *RPCServer, params *RawParams) *msgjson.ResponsePayload {
+	form, err := parseWithdrawArgs(params)
+	if err != nil {
+		return usage(withdrawRoute, err)
+	}
+	defer form.AppPass.Clear()
+	if exists := s.core.WalletState(form.AssetID) != nil; !exists {
```

Comment:
This requires some minor mental gymnastics, especially with the double negative.  I think this could be

```go
if s.core.WalletState(form.AssetID) == nil {
```

or if you want to be more explicit

```go
if state := s.core.WalletState(form.AssetID); state == nil {
```

Reply:
Oh yeah, not that you commented I can see how ugly this is. Going with the first suggestion.